### PR TITLE
Connect tax classes view to API

### PR DESCRIPTION
**Pull request by Droniu**
I want to merge this change because it connects tax classes view to the API in simple taxes feature.

**PR intended to be tested with API branch:** https://github.com/saleor/saleor/pull/9784

### Pull Request Checklist

1. [x] This code contains UI changes
2. [x] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://saleor-6391-simple-taxes.api.saleor.rocks/graphql/
MARKETPLACE_URL=https://marketplace-gray.vercel.app/

### Do you want to run more stable tests?
Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation

CONTAINERS=1

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-6830 Connect to API - Tax classes view](https://app.clickup.com/t/2549495/SALEOR-6830)

File: src/taxes/pages/TaxClassesPage/TaxClassesPage.tsx [hunk 1]
```diff
@@ -24,23 +33,40 @@ import {
...
   handleTabChange: (tab: string) => void;
+  savebarState: ConfirmButtonTransitionState;
+  disabled: boolean;
+  onCreateNewButtonClick: () => void;
+  onTaxClassDelete: (id: string) => SubmitPromise;
+  onTaxClassCreate: (input: TaxClassCreateInput) => SubmitPromise;
+  onTaxClassUpdate: (id: string, input: TaxClassUpdateInput) => SubmitPromise;
 }
 
 export const TaxClassesPage: React.FC<TaxClassesPageProps> = props => {
```
Comment (andrzejewsky) [inline src/taxes/pages/TaxClassesPage/TaxClassesPage.tsx:56]: can you overall split this page into smaller components? each for one responsibility, at leas to some extent, split it.

File: src/taxes/pages/TaxClassesPage/TaxClassesPage.tsx [hunk 2]
```diff
@@ -24,23 +33,40 @@ import {
...
 interface TaxClassesPageProps {
   taxClasses: TaxClassFragment[] | undefined;
   selectedTaxClassId: string;
   handleTabChange: (tab: string) => void;
+  savebarState: ConfirmButtonTransitionState;
+  disabled: boolean;
+  onCreateNewButtonClick: () => void;
+  onTaxClassDelete: (id: string) => SubmitPromise;
+  onTaxClassCreate: (input: TaxClassCreateInput) => SubmitPromise;
+  onTaxClassUpdate: (id: string, input: TaxClassUpdateInput) => SubmitPromise;
```
Comment (andrzejewsky) [inline src/taxes/pages/TaxClassesPage/TaxClassesPage.tsx:53]: that sound like scoped responsibility, you can move out that logic to the dedicated component, avoid props-drilling

**Review by andrzejewsky — APPROVED** [review]
Please look at the comments, and try to avoid building big page components, and entire flow in one place - eg. when you have form, create hooks for handle it, create form itself use together

**Review by orzechdev — APPROVED** [review]

**Pull request merged**